tests/density_estimator_test.py

```diff
@@ -10,59 +10,86 @@
 from torch import eye, zeros
 from torch.distributions import MultivariateNormal
 
-from sbi.neural_nets.density_estimators import NFlowsFlow, ZukoFlow
 from sbi.neural_nets.density_estimators.shape_handling import reshape_to_iid_batch_event
-from sbi.neural_nets.flow import build_nsf, build_zuko_maf
+from sbi.neural_nets.flow import (
+    build_nsf,
+    build_zuko_bpf,
+    build_zuko_cnf,
+    build_zuko_gf,
+    build_zuko_maf,
+    build_zuko_naf,
+    build_zuko_ncsf,
+    # build_zuko_gmm,
+    build_zuko_nice,
+    build_zuko_nsf,
+    build_zuko_sospf,
+    build_zuko_unaf,
+)
+
+
+def get_batch_input(nsamples: int, input_dims: int) -> torch.Tensor:
+    input_mvn = MultivariateNormal(
+        loc=zeros(input_dims), covariance_matrix=eye(input_dims)
+    )
+    return input_mvn.sample((nsamples,))
+
+
+def get_batch_context(nsamples: int, condition_shape: tuple[int, ...]) -> torch.Tensor:
+    context_mvn = MultivariateNormal(
+        loc=zeros(*condition_shape), covariance_matrix=eye(condition_shape[-1])
+    )
+    return context_mvn.sample((nsamples,))
 
 
-@pytest.mark.parametrize("density_estimator", (NFlowsFlow, ZukoFlow))
+@pytest.mark.parametrize(
+    "build_density_estimator",
+    (
+        build_nsf,
+        # build_zuko_gmm,
+        build_zuko_nice,
+        build_zuko_maf,
+        build_zuko_nsf,
+        build_zuko_ncsf,
+        build_zuko_sospf,
+        build_zuko_naf,
+        build_zuko_unaf,
+        build_zuko_cnf,
+        build_zuko_gf,
+        build_zuko_bpf,
+    ),
+)
 @pytest.mark.parametrize("input_dims", (1, 2))
 @pytest.mark.parametrize(
     "condition_shape", ((1,), (2,), (1, 1), (2, 2), (1, 1, 1), (2, 2, 2))
 )
-def test_api_density_estimator(density_estimator, input_dims, condition_shape):
+def test_api_density_estimator(build_density_estimator, input_dims, condition_shape):
```

Comment:
Nice idea to refactor it in terms of the builders! Let's make sure to test all builders then, including all the `NFlowsFlow` ones.